lightning/src/onion_message/messenger.rs

```diff
@@ -659,78 +639,108 @@ where
 			}
 		}
 	}
+}
+
+impl<G: Deref<Target = NetworkGraph<L>>, L: Deref, ES: Deref> MessageRouter
+	for DefaultMessageRouter<G, L, ES>
+where
+	L::Target: Logger,
+	ES::Target: EntropySource,
+{
+	fn find_path(
+		&self, sender: PublicKey, peers: Vec<PublicKey>, destination: Destination,
+	) -> Result<OnionMessagePath, ()> {
+		Self::find_path(&self.network_graph, sender, peers, destination)
+	}
 
-	pub(crate) fn create_blinded_paths<T: secp256k1::Signing + secp256k1::Verification>(
-		network_graph: &G, recipient: PublicKey, context: MessageContext, peers: Vec<PublicKey>,
-		entropy_source: &ES, secp_ctx: &Secp256k1<T>,
+	fn create_blinded_paths<T: secp256k1::Signing + secp256k1::Verification>(
+		&self, recipient: PublicKey, context: MessageContext, peers: Vec<MessageForwardNode>,
+		secp_ctx: &Secp256k1<T>,
 	) -> Result<Vec<BlindedMessagePath>, ()> {
-		let peers =
-			peers.into_iter().map(|node_id| MessageForwardNode { node_id, short_channel_id: None });
 		Self::create_blinded_paths_from_iter(
-			network_graph,
+			&self.network_graph,
 			recipient,
 			context,
 			peers.into_iter(),
-			entropy_source,
+			&self.entropy_source,
 			secp_ctx,
 			false,
 		)
 	}
+}
 
-	pub(crate) fn create_compact_blinded_paths<T: secp256k1::Signing + secp256k1::Verification>(
-		network_graph: &G, recipient: PublicKey, context: MessageContext,
-		peers: Vec<MessageForwardNode>, entropy_source: &ES, secp_ctx: &Secp256k1<T>,
-	) -> Result<Vec<BlindedMessagePath>, ()> {
-		Self::create_blinded_paths_from_iter(
-			network_graph,
-			recipient,
-			context,
-			peers.into_iter(),
-			entropy_source,
-			secp_ctx,
-			true,
-		)
+/// A [`MessageRouter`] that can only route to a directly connected [`Destination`],
+/// and tries to create compact blinded path for that purpose.
```

Comment:
We should note that if the caller passes `create_blinded_paths` with `MessageForwardNode`s that do not contain an scid, then the paths will not be compact.

Reply:
Makes sense!
Updated the docs [pr3246.19](https://github.com/shaavan/rust-lightning/commits/pr3246.19) to properly reflect the best-effort basis nature of `CompactMessageRouter` in [pr3246.19](https://github.com/shaavan/rust-lightning/commits/pr3246.19)
Thanks a lot for the pointer!